supersim_test.go

```diff
@@ -21,57 +20,70 @@ const (
 	l1BlockAddress                    = "0x4200000000000000000000000000000000000015"
 )
 
-func TestGenesisState(t *testing.T) {
-	logger := oplog.NewLogger(os.Stderr, oplog.DefaultCLIConfig())
-	supersim := NewSupersim(logger, &DefaultConfig)
-	err := supersim.Start(context.Background())
-
-	defer func() {
-		err := supersim.Stop(context.Background())
-		if err != nil {
-			t.Fatalf("Failed to stop supersim: %v", err)
+type TestSuite struct {
+	t *testing.T
+
+	Cfg      *Config
+	Supersim *Supersim
+}
+
+func createTestSuite(t *testing.T) *TestSuite {
+	cfg := &DefaultConfig
+	testlog := testlog.Logger(t, log.LevelInfo)
+
+	supersim := NewSupersim(testlog, cfg)
+	t.Cleanup(func() {
+		if err := supersim.Stop(context.Background()); err != nil {
+			t.Errorf("failed to stop supersim: %s", err)
 		}
-	}()
+	})
 
-	if err != nil {
-		t.Fatalf("Failed to start supersim: %v", err)
+	if err := supersim.Start(context.Background()); err != nil {
+		t.Fatalf("unable to start supersim: %s", err)
+		return nil
 	}
 
-	for _, l2ChainConfig := range DefaultConfig.l2Chains {
-		rpcUrl := fmt.Sprintf("http://127.0.0.1:%d", l2ChainConfig.Port)
-		client, clientCreateErr := rpc.Dial(rpcUrl)
+	return &TestSuite{
+		t:        t,
+		Cfg:      cfg,
+		Supersim: supersim,
+	}
+}
 
-		if clientCreateErr != nil {
-			t.Fatalf("Failed to create client: %v", clientCreateErr)
-		}
+func TestStartup(t *testing.T) {
+	testSuite := createTestSuite(t)
+
+	// test that all chains can be queried
+	client, err := rpc.Dial(testSuite.Supersim.l1Chain.Endpoint())
+	require.NoError(t, err)
+	require.NoError(t, client.CallContext(context.Background(), nil, "eth_chainId"))
+	client.Close()
+
+	for _, l2Chain := range testSuite.Supersim.l2Chains {
+		client, err := rpc.Dial(l2Chain.Endpoint())
+		require.NoError(t, err)
+		require.NoError(t, client.CallContext(context.Background(), nil, "eth_chainId"))
```

Comment:
similar comment here, Is it worth checking that the value of the chain id in the response is the same as the chain id in the corresponding l2 chain config?